fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/federation/Challenge.kt

```diff
@@ -19,6 +20,7 @@ class Challenge
 
   init {
     this.validUntil = max(0L, validUntil)
```

Comment:
validation should ideally be before instantiating attributes, since if the input is wrong and it is meant to crash, no need to fill the attribute beforehand.